hawtio-system/src/main/java/io/hawt/web/FileUploadFilter.java

```diff
@@ -0,0 +1,171 @@
+
+                    addFilters(filter);
+                }
+            } catch (RuntimeException e) {
+                LOG.warn("Error configuring filter {}", config);
+            }
+        }
+    }
+
+    private static byte[] hexStringToByteArray(String s) {
+        int len = s.length();
+        byte[] data = new byte[len / 2];
+        for (int i = 0; i < len; i += 2) {
+            data[i / 2] = (byte) ((Character.digit(s.charAt(i), 16) << 4)
+                + Character.digit(s.charAt(i + 1), 16));
+        }
+        return data;
+    }
+
+    // Exposed for testing
+    public long translateFileSize(String size) {
+
+        if (size.toLowerCase().trim().contains("bytes")) {
+            String normalized = size.toLowerCase().replace("bytes", "").trim();
+            return Long.parseLong(!normalized.equals("") ? normalized : "0");
+        }
+
+        if (size.toLowerCase().trim().contains("kb")) {
+            String normalized = size.toLowerCase().replace("kb", "").trim();
+            return 1024L * Long.parseLong(!normalized.equals("") ? normalized : "0");
+        }
+
+        if (size.toLowerCase().trim().contains("mb")) {
+            String normalized = size.toLowerCase().replace("mb", "").trim();
+            return 1048576L * Long.parseLong(!normalized.equals("") ? normalized : "0");
+        }
+
+        if (size.toLowerCase().trim().contains("gb")) {
+            String normalized = size.toLowerCase().replace("gb", "").trim();
+            return 1073741824L * Long.parseLong(!normalized.equals("") ? normalized : "0");
+        }
+
+        return 0L;
+    }
+
+    public boolean accept(byte[] fileContent) {
+        if (this.getFilters().isEmpty()) {
+            return true;
+        }
+
+        boolean fileAccepted = false;
+        for (MagicNumberFileFilter magicNumberFileFilter : this.getFilters()) {
+
+            if (fileContent.length >= magicNumberFileFilter.getByteOffset() +
+                magicNumberFileFilter.getMagicNumbers().length) {
+                byte[] fileMagicBytes = Arrays.copyOfRange(fileContent, toIntExact(magicNumberFileFilter.getByteOffset()),
+                    toIntExact(magicNumberFileFilter.getByteOffset()) + magicNumberFileFilter.getMagicNumbers().length);
+                if (Arrays.equals(magicNumberFileFilter.getMagicNumbers(), fileMagicBytes)) {
+                    //Checking the file size
+                    String fileSize = FileUtils.byteCountToDisplaySize(fileContent.length);
+
+                    if (translateFileSize(fileSize) <= translateFileSize(magicNumberFileFilter.getMaxSize())) {
+                        fileAccepted = true;
+                    }
+                }
+            }
+        }
+
+        return fileAccepted;
+    }
+
+    public List<MagicNumberFileFilter> getFilters() {
+        if (this.filters == null) {
+            this.filters = new ArrayList<>();
+        }
+
+        return filters;
+    }
+
+
+    static final class MagicNumberFileFilter {
```

Comment:
Why is it package private?  The `getFilters` method being public, doesn't it cause any problems when an user tries to add a custom filter from outside of the package?

Reply:
They are not suppose to, but you're right filter list should be immutable list.

Reply:
Ah yes making it immutable is also a good practice. What I meant though is that if `MagicNumberFileFilter` is package private, the class should not be visible from external packages. So I wonder what would happen when an user code tries to invoke the `getFilters` method from, say, `com.example.app`?

Reply:
They are not suppose to, I've changed it to `private` and made the list immutable. Essentially this filter is a global filter that only allows user to access the global file filter by invoking `FileUploadFilter fileUploadFilter = FileUploadFilter.newFileUploadFilter()` which gives them access to the global filter by invoking the `getFilters` which is an immutable list of filters. This can be used throughout the application to filter any file upload.